### DISCUSSION TITLE
Does Firecrawl Offer Link Connection Tracking?

### DISCUSSION THREAD

**Discussion by AIB0I** [firecrawl/firecrawl#667] [Q&A]
I am looking to build a link graph where nodes represent links (URLs) and edges represent connections between these links. Specifically, I want to create a graph where an edge exists between two nodes if one link is mentioned within another link. For example, if link2 is mentioned in link1, there should be a connection (edge) between link1 and link2.

Does Firecrawl provide any way to track these connections using the "crawl" or "map" API?

**Comment by 173787247** [comment]
## Files / crawl / PDF («Does Firecrawl Offer Link Connection Tracking?»)

- **PDF + CJK**: need an embeddable font / a backend that keeps unicode; “garbled markdown” is usually extraction encoding, not your editor.
- **Crawl**: respect robots, cap depth, and don’t dump full HTML into the LLM — markdown + metadata, or you blow the context.
- **Memory**: Playwright/Chromium crawlers idle-high; reuse one browser, don’t spawn per URL.
